### PR TITLE
Merge all profiling-related crates into a single `datadog-profiling` crate

### PR DESCRIPTION
**Pull request by ivoanjo**
# What does this PR do?

This PR takes the previously-existing profiling-related crates:

* `ddprof`
* `ddprof-exporter`
* `ddprof-profiles`

and folds them into a single `datadog-profiling` crate.

**Note that I did not touch the `ddprof-ffi` crate yet. To be decided what do do with it**.

# Motivation

This follows from our discussions on reducing the number of crates during PRs and the weekly meeting.

# Additional Notes

I kept the existing namespacing, in a way:

* `ddprof-exporter` -> `datadog_profiling::exporter`
* `ddprof-profiles` -> `datadog_profiling::profile`

Also, this PR will be really hard to rebase and whatnot, so I suggest any extra adjustments be made as a separate, follow-up PR, rather than letting this live for a long time. (I'll still be up to doing any changes, definitely not saying this to avoid doing any further work)

I also went ahead and reserved the `datadog-profiling` name on crates.io => https://crates.io/crates/datadog-profiling .

# How to test the change?

* `cargo test`
* run test applications

### CODE REVIEW

**Review by pawelchcki — APPROVED** [review]
lgtm

File: ddprof-ffi/cbindgen.toml [hunk 1]
```diff
@@ -28,4 +28,4 @@ must_use = "DDOG_CHECK_RETURN"
 
 [parse]
 parse_deps = true
-include = ["ddcommon", "ddcommon-ffi", "ddprof-exporter", "ddprof-profiles", "ux"]
+include = ["ddcommon", "ddcommon-ffi", "datadog-profiling", "ux"]
```
Comment (r1viollet) [inline ddprof-ffi/cbindgen.toml:31]: Thanks, this makes sense.

**Review by r1viollet — APPROVED** [review]
LGTM

**Comment by morrisonlevi** [comment]
Why `ddprof-profiles -> datadog_profiling::profile` instead of `datadog_profiling::profiles`? Not asking you to change it, just trying to understand why it wasn't kept in the plural.

**Comment by ivoanjo** [comment]
> Why ddprof-profiles -> datadog_profiling::profile instead of datadog_profiling::profiles? Not asking you to change it, just trying to understand why it wasn't kept in the plural.

Good question. I felt it was a bit more in-line with the rest of the names:

* we have "exporter", not "exporters"
* we may at some point have "ffi", not "ffis"
* it seems a bit more natural to talk about profile::Mapping than profiles::Mapping

...but if you feel strongly against it, happy to change -- I did choose between both explicitly, and I do prefer singular, but I'm willing to compromise with profiles :)

**Review by morrisonlevi — APPROVED** [review]

File: profiling/build.rs [hunk 2]
```diff
@@ -15,8 +15,11 @@ fn main() -> Result<(), std::io::Error> {
         .type_attribute("pprof.Sample", "#[derive(Eq, Hash)]")
         .type_attribute("pprof.ValueType", "#[derive(Copy, Eq, Hash)]");
 
-    let protos = &[concat!(env!("CARGO_MANIFEST_DIR"), "/src/profile.proto")];
-    let includes = &[concat!(env!("CARGO_MANIFEST_DIR"), "/src")];
+    let protos = &[concat!(
+        env!("CARGO_MANIFEST_DIR"),
+        "/src/profile/profile.proto"
+    )];
```
[the remark this answers is not in the document]
Comment (ivoanjo) [reply profiling/build.rs:21]: Yes, it did. I don't particularly like some of its decisions but... ahaha not going to fight that fight xD

**Comment by ivoanjo** [comment]
Thanks everyone for the quick turnaround :)

**Pull request merged**